src/samples/network/netctl.c

```diff
@@ -94,8 +97,53 @@ error_cb(void *data, const struct sol_netctl_service *service,
 }
 
 static void
+report_error(void *data, struct sol_netctl_service *service,
+    const char *error)
+{
+    int r;
+
+    printf("The agent action error is %s\n", error);
+    r = sol_netctl_report_error(service, SOL_NETCTL_AGENT_NORETRY);
+    printf("The agent report error return value is %d\n", r);
+}
+
+static void
+request_input(void *data, struct sol_netctl_service *service,
+    const struct sol_ptr_vector *vector)
+{
+    int r, i = 0;
+    struct sol_netctl_agent_input *input;
+
+    printf("The agent action is input\n");
+
+    SOL_PTR_VECTOR_FOREACH_IDX (vector, input, i) {
+        printf("The agent input type is %d\n", input->type);
+        input->input = strdup(INPUT);
```

Comment:
Yes, strdup need to be checked.

>Why are you doing this strdup?
input->input will be used and freed in "sol_netctl_request_input".
Therefore, "strdup" is used for input->input and input->input need be freed in "sol_netctl_request_input".